hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,46 @@
+if answers['katello']
+  MONGO_DB_NAME = param('katello', 'pulp_db_name')
+else
+  MONGO_DB_NAME = param('foreman_proxy_content', 'pulp_db_name')
+end
+
+def migration
+  export_dir = '/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+
+  # Create directories and dump MongoDB
+  logger.info 'Stopping services except MongoDB'
+  Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
```

Comment:
What if my `mongod` service is not from SCL or maybe even a different SCL version? In fact AFAKT the installer doesn't even support SCL https://github.com/theforeman/puppet-pulp/blob/master/manifests/database.pp

Reply:
We fixed that by adding mongo-syspaths so we can just call mongo, here is the pr:

https://github.com/theforeman/foreman-packaging/pull/2587